src/azul/plugins/repository/tdr/__init__.py

```diff
@@ -241,10 +241,12 @@ def _stitch_bundles(self,
                 unprocessed |= self._find_upstream_bundles(dangling_inputs) - processed
             else:
                 log.debug('Bundle %r is self-contained', bundle)
-        if log.isEnabledFor(logging.DEBUG):
-            log.debug('Stitched together bundles: %r', processed)
-        else:
-            log.info('Stitched together %i bundles', len(processed))
+        processed.remove(root_bundle)
+        if processed:
+            if log.isEnabledFor(logging.DEBUG):
+                log.debug('Stitched bundles: %r', processed)
```

Comment:
```suggestion
                log.debug('Stitched %i bundles: %r', len(processed), processed)
```

and think about ways to reduce the resulting code duplication.